### DISCUSSION TITLE
Change image proportions possible?

### DISCUSSION THREAD

**Discussion by ilia239** [FDH2/UxPlay#348] [General]
Hello,
Is it possible to adjust width/height proportions of an image before streaming it from the device?
The problem I am trying to solve - a picture of a perfect square streamed to my Epson streamer looks like a rectangular, the width is slightly longer than length, proportions are not correct . As I am using the setup for drawing, even small disproportions are not acceptable. I am looking for a way to fix width/height proportions of the final image on the screen.
My Epson streamer does not have an option to shrink width or height, which would be most convenient solution indeed, so I need an image pre-processing: to scale the 1080 x 720 image to i.e. 1055 x 720 and fill the left and right space with black
How can I do this with UxPlay? What other options do I have?
regards,
ilia

**Comment by fduncanh** [comment]
The video pipeline has a "videoscale" in it, but there isnt  a uxplay  startup option to change it. (it could be added)

You can edit the pipeline in lib/video_renderer.c   (used to be called video_renderer_gstreamer.c)

the default pipeline is 
appsrc name=video_source ! queue ! h264parse ! decodebin ! videoconvert ! videoscale ! autovideosink name=autovideosink_h264 sync=true"

I found an example of a pipeline doing this:
https://www.technomancy.org/gstreamer/scaling-a-video/

see
https://gstreamer.freedesktop.org/documentation/videoconvertscale/videoscale.html?gi-language=c
